hw/misc/hermes.c

```diff
@@ -248,6 +279,88 @@ static void hermes_bar0_write(void *opaque, hwaddr addr, uint64_t val,
     hermes_bar_warn_read_only(0, addr);
 }
 
+static int hermes_execute_descs(HermesState *hermes,
+                                struct hermes_bar2_dir_reg *dir,
+                                hwaddr *desc_addr, uint8_t *num_desc)
+{
+    struct hermes_dma_desc *desc;
+    int i;
+
+    trace_hermes_dma_num_adj(*num_desc - 1);
+    desc = g_new(struct hermes_dma_desc, *num_desc);
+    if (!desc) {
+        fprintf(stderr, "[Hermes] Failed to alloc memory for DMA descriptor\n");
+        return -1;
+    }
+
+    /* Read DMA descriptors */
+    pci_dma_read(&hermes->pdev, *desc_addr, desc, *num_desc * sizeof(*desc));
+    for (i = 0; i < *num_desc; i++) {
```

Comment:
What's the point of putting all the traces in their own loop?

Reply:
Ops, missed that